### ISSUE TITLE
K-Nearest Neighbors (KNN)

### ISSUE THREAD

**Issue by Janecxl** [Janecxl/Notes#7]
## **What is KNN 🤔**
**KNN** works by finding the distances between a query and all the input data, selecting the closest K neighbors to the query. Then vote for the most frequent label (classification) or average the labels (regression).
 KNN assumes similar things exist closely.

## **Pros and Cons**
✅ Little training effort
❌ Since we use the training data itself as part of the querying process, things can get slow and unwieldy (in both time and space).

## **Effect of K**

- Small K: Very flexible. May overfit.
- Large K: Inflexible. Smooth boundary.

- How to choose K: [Cross-validation](https://github.com/Janecxl/Notes/issues/18).

## **Reference 📖**
[Machine Learning Basics with the K-Nearest Neighbors Algorithm](https://towardsdatascience.com/machine-learning-basics-with-the-k-nearest-neighbors-algorithm-6a6e71d01761)
Cornell ORIE4740: Statistical Data Mining slides
Gatech CS7641: [Machine Learning notes](https://teapowered.dev/assets/ml-notes.pdf)
